### ISSUE TITLE
performance?

### ISSUE THREAD

**Issue by ahihi** [rncm-prism/prism-samplernn#14]
hello,

do you have any ballpark performance numbers for training and generation? i've started my own training session so i'll find out eventually, but it would be nice to have a rough idea of how long i should expect things to take!

**Comment by relativeflux** [comment]
Hi,

It's quite difficult to give a good answer to that, since the training time depends on a number of factors which will typically vary widely from case to case. For example factors relating to input, such as dataset size (number of audio chunks), sample rate, and also internal factors such as dimensionality of the network, number of layers, etc. And the variation in these is not necessarily linear. Also remember that there is more than a single metric by which to judge the 'speed' of training - the raw duration of each step / batch / epoch (these stats are reported continuously during training, so it's possible to make a crude estimate), but also the rate at which the network converges towards a 'good' set of predictions.

With a dataset of a few thousand files at 16k sr I have rarely trained for less than a day. That's using a single Titan RTX.

Hope that's of some help.

**Comment by ahihi** [comment]
yes that helps me a lot :) thank you!

btw, do you have/know any publicly available pretrained model that i could try out in the meantime?

**Comment by relativeflux** [comment]
No problem. Unfortunately I don't know of any public models, we did release one back in June based on a large choral dataset but it's based on deprecated hyperparams so won't won't now. We are planning to release a number of new models in the coming months, however.

We have been getting nice results with LSTM + linear quantization + skip connections, if that's any help.
